include/openrave/kinbody.h

```diff
@@ -330,6 +332,39 @@ class OPENRAVE_API KinBody : public InterfaceBase
         bool _bVisible = true; ///< if true, geometry is visible as part of the 3d model (default is true)
         bool _bModifiable = true; ///< if true, object geometry can be dynamically modified (default is true)
 
+        struct CalibrationBoardParameters { ///< used by GT_CalibrationBoard
+            int numDotsX; ///< number of dots in x direction, minimum 3
+            int numDotsY; ///< number of dots in y direction, minimum 3
+            dReal dotsDistanceX; ///< distance between center of dots in x direction
+            dReal dotsDistanceY; ///< distance between center of dots in y direction
+            RaveVector<dReal> dotColor; ///< color of dot mesh, which can differ from the color of the board mesh
+            std::string patternName; ///< string that determines pattern of big and normal dots
+                                     /// currently, the only supported type is "threeBigDotsDotGrid"
+                                     /// any unsupported pattern will generate a full grid of normal dots
+            dReal dotDiameterDistanceRatio; ///< dot diameter divided by minimum of dot distances x and y
+            dReal bigDotDiameterDistanceRatio; ///< big dot diameter divided by minimum of dot distances x and y
+            ///< trimesh representation of the collision data of calibration board dot grid in this local coordinate system
+            /// Shares the same caveats as _meshcollision.
+            TriMesh _dotmeshcollision;
+            bool operator==(const CalibrationBoardParameters& other) const {
+                return numDotsX == other.numDotsX
+                       && numDotsY == other.numDotsY
+                       && dotsDistanceX - other.dotsDistanceX < dotsDistanceX*g_fEpsilon
+                       && dotsDistanceY - other.dotsDistanceY < dotsDistanceY*g_fEpsilon
+                       && dotColor == other.dotColor
+                       && patternName == other.patternName
+                       && dotDiameterDistanceRatio - other.dotDiameterDistanceRatio < dotDiameterDistanceRatio*g_fEpsilon
+                       && bigDotDiameterDistanceRatio - other.bigDotDiameterDistanceRatio < bigDotDiameterDistanceRatio*g_fEpsilon;
+            }
+            bool operator!=(const CalibrationBoardParameters& other) const {
+                return !operator==(other);
+            }
+        };
+        CalibrationBoardParameters _calibrationBoardParams;
```

Comment:
- Unfortunately the current structure will add calibration parameters to every GeometryInfo, even ones that do not have it. If you can encode this in a way that would add minimal size to GeometryInfo, perhaps it is OK (For example encoding most of the parameters inside `_vGeomDataX`. I do think having `CalibrationBoardParameters` as the way you define it is OK, but the way we insert it into GeometryInfo could be better. For example, we can make it a pointer (but then it becomes tricky, one interesting way out of this is to use `std::vector<CalibrationBoardParameters>` and resize it to 1 whenever the geomtype is `GT_CalibrationBoard`.

Reply:
Something like `_vSideWalls` I guess?

```cpp
std::vector<SideWall> _vSideWalls; ///< used by GT_Cage
```